client/src/placement-context.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useEffect, useRef } from "react";
+import { useGleanClick } from "./telemetry/glean-context";
+import useSWR from "swr";
+import { PLACEMENT_ENABLED } from "./env";
+import { useUserData } from "./user-context";
+import { useLocation } from "react-router";
+
+export enum Status {
+  success = "success",
+  geoUnsupported = "geo_unsupported",
+  capReached = "cap_reached",
+}
+
+export interface Fallback {
+  click: string;
+  view: string;
+  copy: string;
+  image: string;
```

Comment:
Just checking: `copy` and `image` here are not optional, but they are in `PlacementStatus`?

Reply:
yes